geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -1201,13 +1201,14 @@ protected void addRemovedEvent(PartitionedRegion prQ, int bucketId, Object key)
 
   public void sendQueueRemovalMesssageForDroppedEvent(PartitionedRegion prQ, int bucketId,
       Object key) {
-    final HashMap<String, Map<Integer, List>> temp = new HashMap<String, Map<Integer, List>>();
-    Map bucketIdToDispatchedKeys = new ConcurrentHashMap();
-    temp.put(prQ.getFullPath(), bucketIdToDispatchedKeys);
-    addRemovedEventToMap(bucketIdToDispatchedKeys, bucketId, key);
     Set<InternalDistributedMember> recipients =
-        removalThread.getAllRecipients(sender.getCache(), temp);
+        removalThread.getAllRecipientsForEvent(sender.getCache(), prQ.getFullPath(), bucketId);
+
     if (!recipients.isEmpty()) {
+      final HashMap<String, Map<Integer, List>> temp = new HashMap<String, Map<Integer, List>>();
+      Map bucketIdToDispatchedKeys = new ConcurrentHashMap();
```

Comment:
Try to make sure all collections are typed `Map<something, something>`.

Reply:
updated